src/Sources/Code/Module.cpp

```diff
@@ -234,71 +276,125 @@ void WriteSimVar(SimVar& simVar, Client* client) {
 		sizeof(simVar.Value),
 		&simVar.Value
 	);
-	if (hr != S_OK) {
-		fprintf(stderr, "MobiFlight[%s]: Error on Setting Client Data. %u, SimVar: %s (ID: %u)", client->Name.c_str(), hr, simVar.Name.c_str(), simVar.ID);
-	}
+
+		if (hr != S_OK) {
+			fprintf(stderr, "MobiFlight[%s]: Error on Setting Client Data. %lu, SimVar: %s (ID: %u)", client->Name.c_str(), hr, simVar.Name.c_str(), simVar.ID);
+		}
 #if _DEBUG
-	std::cout << "MobiFlight[" << client->Name.c_str() << "]: SimVar " << simVar.Name.c_str();
+	std::cout << "MobiFlight[" << client->Name.c_str() << "]: Written SimVar " << simVar.Name.c_str();
 	std::cout << " with ID " << simVar.ID << " has value " << simVar.Value << std::endl;
 #endif
 }
 
-// Register a single SimVar and send the current value to SimConnect Clients 
-void RegisterSimVar(const std::string code, Client* client) {
+// Register a single SimVar and send the current value to SimConnect Clients
+void RegisterSimVar(const std::string code, Client* client, const bool isString) {
 	std::vector<SimVar>* SimVars = &(client->SimVars);
-	SimVar var1;
-	var1.Name = code;
-	var1.ID = SimVars->size() + client->DataDefinitionIdSimVarsStart;
-	var1.Offset = (SimVars->size()) * sizeof(float);
-
-	SimVars->push_back(var1);
+	std::vector<StringSimVar>* StringSimVars = &(client->StringSimVars);
+	SimVar newSimVar;
+	StringSimVar newStringSimVar;
 	HRESULT hr;
 
-	if (client->MaxClientDataDefinition < SimVars->size())
-	{
-		hr = SimConnect_AddToClientDataDefinition(
-			g_hSimConnect,
-			var1.ID,
-			var1.Offset,
-			sizeof(float),
-			0
-		);
 
-		client->MaxClientDataDefinition = SimVars->size();
+	if(isString) {
+		newStringSimVar.Name = code;
+		newStringSimVar.ID = StringSimVars->size() + client->DataDefinitionIdStringVarsStart;
+		newStringSimVar.Offset = StringSimVars->size() * MOBIFLIGHT_STRING_SIMVAR_VALUE_MAX_LEN;
+		newStringSimVar.Value.empty();
+		StringSimVars->push_back(newStringSimVar);
+	} else {
+		newSimVar.Name = code;
+		newSimVar.ID = SimVars->size() + client->DataDefinitionIdSimVarsStart;
+		newSimVar.Offset = SimVars->size() * (sizeof(float));
+		newSimVar.Value = 0.0F;
```

Comment:
Oh boy, that took some time of my life. Some of my testing code failed and I did not understand why.
Before that initialization change, each new Registered SimVar reliably generated 2 callbacks after registration (because of the 2 WriteSimVar Calls later on). Now many SimVars only generate 1 callback. All SimVars with a real current value of 0.0F now only generate 1 callback instead of previously 2. 
-> So that code change in fact changes the initial callback behavior to the clients.
-> To stay 100% compatible the variable should not be initialized as before. 
BUT
I would propose instead now to do it really properly (small incompatible) and take the chance to remove the strange (unecessary?) first WriteSimVar before execute_calculator_code. If Sebastian cannot remember any reason for it, we should remove it now. This is a small incompatible change. Then not each float variable will get an initial 0.0 callback before the real value. For being able to do such small changes, I introduced the versioning interface, so clients can detect the slightly different behavior, if they want to.

Reply:
Okay so I removed the first WriteSimVar and kept the initializiation